packages/trader/src/Stores/Modules/Trading/trade-store.ts

```diff
@@ -1386,6 +1398,14 @@ export default class TradeStore extends BaseStore {
             this.setMarketStatus(isMarketClosed(this.active_symbols, obj_new_values.symbol ?? ''));
             has_only_forward_starting_contracts =
                 ContractType.getContractCategories().has_only_forward_starting_contracts;
+            // Set stake to default one (from contracts_for) on symbol switch
+            if (this.default_stake && this.is_dtrader_v2_enabled) {
```

Comment:
Thanks , it works like a charm

Reply:
im trying to understand why this needs to be done inside processNewValuesAsync. can we  

onChange({ value: this.default_stake ?? '', name: 'stake'}) to replace obj_new_value.amount = this.default_stake inside stake? Same thing above also.

then it'll be combination of onChange and setV2ParamsInitialValue in stake itself based on given conditions.

this way default_stake doesnt even have to get stored in trade-store.

everything you need will live in usecontractsforcompany and stake